src/meta/src/barrier/mod.rs

```diff
@@ -733,18 +774,52 @@ impl GlobalBarrierManager {
             panic!("failed to execute barrier: {}", err.as_report());
         }
     }
+}
 
+impl GlobalBarrierManagerContext {
     /// Try to commit this node. If err, returns
-    async fn complete_barrier(&mut self, node: &mut EpochNode) -> MetaResult<()> {
-        let prev_epoch = node.command_ctx.prev_epoch.value().0;
-        match &mut node.state {
-            Completed(resps) => {
+    async fn complete_barrier(self, node: EpochNode) -> MetaResult<bool> {
```

Comment:
The diff looks not friendly to reviewers since it's apart from the original `Manager::handle_barrier_complete`. 🤣